site/app/libraries/database/DatabaseQueries.php

```diff
@@ -528,6 +528,11 @@ public function setAnnouncement($thread_id, $onOff){
         $this->course_db->query("UPDATE threads SET pinned = ? WHERE id = ?", array($onOff, $thread_id));
     }
 
+    public function getNumberOfPinnedThreadsAndAttachments() {
+        $this->course_db->query("SELECT COUNT(*) FROM threads WHERE pinned=true");
```

Comment:
We might want to add an addition `WHERE` clause to check if the announcement threads are a week old before notifying the instructor.